test/Datadog.Trace.ClrProfiler.Managed.Tests/Reflection/ModuleLookupTests.cs

```diff
@@ -0,0 +1,72 @@
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using Datadog.Trace.ClrProfiler.Emit;
+using Xunit;
+
+namespace Datadog.Trace.ClrProfiler.Managed.Tests
+{
+    public class ModuleLookupTests
+    {
+        [Fact]
+        public void Lookup_SystemData_Succeeds_WithTwentyThreads()
+        {
+            var threads = new List<Thread>();
+
+            var systemDataGuid = typeof(System.Data.DataTable).Assembly.ManifestModule.ModuleVersionId;
+
+            var bag = new ConcurrentBag<Module>();
+
+            for (var i = 0; i < 20; i++)
+            {
+                threads.Add(new Thread(thread =>
+                {
+                    bag.Add(ModuleLookup.Get(systemDataGuid));
+                }));
+            }
+
+            System.Threading.Tasks.Parallel.ForEach(threads, t => t.Start());
```

Comment:
_Only consider this if it is important to start the threads as close as possible to "at the same time." Otherwise, ignore this comment ;)_

This line is creating 20 parallel tasks (potentially 20 threads) and each one is only executing a single line of code: start _another_ thread. Seems like a lot of extra resources to try starting all 20 threads at the same time. I don't think it works very well, either, since only something like {number of logical cpu cores} tasks will actually run in parallel. I think you might as well do this in a simple loop.

A better approach would be to start all the threads and have them all block waiting for a signal, like a `ManualResetEventSlim `. Once all threads are started and blocked, set the state of the event to signaled so they can all unblock.